src/org/zaproxy/zap/extension/exportreport/Export/ExportReport.java

```diff
@@ -0,0 +1,247 @@
+import javax.swing.SwingWorker;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.extension.ViewDelegate;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.view.View;
+import org.zaproxy.zap.extension.exportreport.ExtensionExportReport;
+import org.zaproxy.zap.extension.exportreport.FileChooser.FileList;
+import org.zaproxy.zap.extension.exportreport.FileChooser.ReportFileView;
+import org.zaproxy.zap.extension.exportreport.FileChooser.ReportFilter;
+import org.zaproxy.zap.extension.exportreport.FileChooser.Utils;
+import org.zaproxy.zap.utils.DesktopUtils;
+import org.zaproxy.zap.view.widgets.WritableFileChooser;
+
+/*
+ * AUTHOR		:	GORAN SARENKAPA - JordanGS
+ * SPONSOR		:	RYERSON UNIVERSITY
+ */
+
+public class ExportReport {
+	private Logger logger = Logger.getLogger(ExportReport.class);
+
+	private Task task; // Create a task and dump the export into the background,
+						// essential for large exports.
+	private WritableFileChooser fc; // Global because the property change
+									// listener won't be able to access if it's
+									// local scope.
+
+	class Task extends SwingWorker<Void, Void> {
+		private ViewDelegate view;
+		private ExtensionExportReport extension;
+		private String fileExtension;
+
+		public Task(ViewDelegate view, ExtensionExportReport extension, String fileExtension) {
+			this.view = view;
+			this.extension = extension;
+			this.fileExtension = fileExtension;
+		}
+
+		/**
+		 * Main task. Executed in background thread.
+		 */
+		@Override
+		public Void doInBackground() {
+			File f = fc.getSelectedFile();
+
+			String file = f.getName();
+			String absolutePath = f.getAbsolutePath();
+			String fileName = file.substring(0, file.lastIndexOf(fileExtension) - 1);
+			String path = absolutePath.substring(0, absolutePath.lastIndexOf(file));
+
+			String XML_PATH = "";
+			try
+			{
+				XML_PATH = ReportExport.generateDUMP(path, fileName, extension.extensionGetTitle(),
+						extension.extensionGetBy(), extension.extensionGetFor(), extension.extensionGetScanDate(),
+						extension.extensionGetScanVer(), extension.extensionGetReportDate(),
+						extension.extensionGetReportVer(), extension.getDescription(),
+						extension.getIncludedAlertSeverity(), extension.getIncludedAlertDetails());
+			}
+			catch (UnsupportedEncodingException e)
+			{
+				logger.error(e.getMessage(), e);
+				view.showWarningDialog(Constant.messages.getString("exportreport.message.error.dump"));
+			}
+
+			String extensionPath = Constant.getZapHome() + "xml" + File.separator;
+			String mergeXSL = (extensionPath + "merge.xml.xsl");
+			String reportXSL = (extensionPath + "report.html.xsl");
+
+			File f_view = null;
+			boolean show = false;
+			switch (fileExtension.toLowerCase(Locale.ROOT)) {
+			case Utils.html:
+				f_view = ReportExport.TRANSFORMATION(view, logger, absolutePath, (path + fileName + ".xml"), XML_PATH, mergeXSL);
```

Comment:
- You don't really need to, you can ignore the returned value if you are not going to use it (as is this case).
 - It's a different (thus custom) logger that will be used in the actual logging instead of the one in the ReportExport class (which does have one ;), even if it didn't have one we should add one to log the messages with the expected class.